server/utils/analytics.test.ts

```diff
@@ -0,0 +1,43 @@
+import { makeChartPalette } from './analytics'
+
+describe.each([
+  [
+    ['Basic', 'Standard', 'Enhanced'],
+    ['app-chart-colour--a', 'app-chart-colour--b', 'app-chart-colour--c'],
+  ],
+  [
+    ['Basic', 'Standard', 'Enhanced', 'Enhanced 2'],
+    ['app-chart-colour--a', 'app-chart-colour--b', 'app-chart-colour--c', 'app-chart-colour--d'],
+  ],
+  [
+    ['Standard', 'Enhanced', 'Enhanced 2'],
+    ['app-chart-colour--b', 'app-chart-colour--c', 'app-chart-colour--d'],
+  ],
+  [
+    ['Basic', 'Standard', 'Enhanced 2'],
+    ['app-chart-colour--a', 'app-chart-colour--b', 'app-chart-colour--d'],
+  ],
+  [
+    ['Basic', 'Standard', 'Enhanced', 'Super enhanced'],
+    ['app-chart-colour--a', 'app-chart-colour--b', 'app-chart-colour--c', 'app-chart-colour--d'],
+  ],
+  [
+    ['Basic', 'Standard', 'Enhanced', 'Enhanced 2', 'Entry'],
+    ['app-chart-colour--a', 'app-chart-colour--b', 'app-chart-colour--c', 'app-chart-colour--d', 'app-chart-colour--e'],
+  ],
+  [
+    ['Basic', 'Standard', 'Enhanced', 'Enhanced 2', 'A', 'B'],
```

Comment:
Is this test validating that 2 unexpected column names are defaulted to `app-chart-colour--e`?

Reply:
i'll add explanations…